### ISSUE TITLE
The Facebook access token is invalid. Please click on Facebook Ads Extension, Manage Settings, go to Advanced options and click on Update token.

### ISSUE THREAD

**Issue by itshaman** [facebookincubator/Facebook-For-OpenCart#91]
Plugin Version: 2.1.3
Try to find at Advanced options  "Update token" button or link, nothing ((

![image](https://user-images.githubusercontent.com/16690535/51936673-cafef680-2419-11e9-9c12-b18fc23d3545.png)
I have button "Delete Settings", 
In my experience, the store setting is erased and it is recreated again. How to fix this problem?
![image](https://user-images.githubusercontent.com/16690535/51936979-790aa080-241a-11e9-993c-acaff59129bc.png)

**Comment by itshaman** [comment]
same problem on another site Opencart v2.3

**Comment by shue249** [comment]
@itshaman Can you send us the Log file for the Facebook Ads Extension here?

![image](https://user-images.githubusercontent.com/11957585/51969809-4f03bd80-24b0-11e9-856a-bb0feebd9c91.png)

**Comment by itshaman** [comment]
`2019-01-16 2:10:22 - Get Sample product feed
2019-01-16 2:10:22 - Complete - Get Sample product feed
2019-01-16 2:10:28 - Updating FAE settings - facebook_enable_cookie_bar
2019-01-16 2:10:28 - Complete - Updating FAE settings
2019-01-16 2:17:38 - Get Sample product feed
2019-01-16 2:17:38 - Complete - Get Sample product feed
2019-01-16 2:17:41 - Get Sample product feed
2019-01-16 2:17:41 - Complete - Get Sample product feed
2019-01-16 2:18:30 - Get Sample product feed
2019-01-16 2:18:30 - Complete - Get Sample product feed
2019-01-16 4:00:58 - Get Sample product feed
2019-01-16 4:00:58 - Complete - Get Sample product feed
2019-01-16 4:01:00 - Get Sample product feed
2019-01-16 4:01:00 - Complete - Get Sample product feed
2019-01-16 4:01:02 - Get Sample product feed
2019-01-16 4:01:02 - Complete - Get Sample product feed
2019-01-16 4:01:04 - Get Sample product feed
2019-01-16 4:01:04 - Complete - Get Sample product feed
2019-01-16 4:01:55 - Updating FAE settings - facebook_catalog_id
2019-01-16 4:01:55 - Complete - Updating FAE settings
2019-01-16 4:01:55 - Updating FAE settings - facebook_dia_setting_id
2019-01-16 4:01:55 - Complete - Updating FAE settings
2019-01-16 4:01:55 - Updating FAE settings - facebook_pixel_id,facebook_pixel_use_pii
2019-01-16 4:01:55 - Complete - Updating FAE settings
2019-01-16 4:01:55 - Clear all Facebook products
2019-01-16 4:01:55 - Complete - Clear all Facebook products
2019-01-16 4:01:55 - Updating FAE settings - facebook_page_id,facebook_page_token
2019-01-16 4:01:55 - Complete - Updating FAE settings
2019-01-16 4:01:55 - Sync all products using feed
2019-01-16 4:01:55 - Generating product feed file
2019-01-16 4:01:55 - Generating product feed file header
2019-01-16 4:01:55 - Generating product feed file for batch 0
2019-01-16 4:01:55 - Generating product feed file for batch 1
2019-01-16 4:01:55 - Sync all products using feed, feed file generated
2019-01-16 4:01:56 - Sync all products using feed, facebook feed created
2019-01-16 4:01:58 - Sync all products using feed, facebook upload created
2019-01-16 4:01:58 - Complete - Sync all products using feed
2019-01-28 21:12:04 - Get Sample product feed
2019-01-28 21:12:04 - Complete - Get Sample product feed
2019-01-28 21:21:53 - Get Sample product feed
2019-01-28 21:21:53 - Complete - Get Sample product feed
2019-01-28 21:22:00 - Facebook Ads Extension version = 2.1.3
2019-01-28 21:22:00 - OpenCart version = 2.3.0.2
2019-01-28 21:22:00 - PHP version = 7.1.20
2019-01-28 22:39:08 - Get Sample product feed
2019-01-28 22:39:08 - Complete - Get Sample product feed
2019-01-29 23:00:03 - Get Sample product feed
2019-01-29 23:00:03 - Complete - Get Sample product feed
2019-01-29 23:00:15 - Get Sample product feed
2019-01-29 23:00:15 - Complete - Get Sample product feed
2019-01-29 23:11:39 - Facebook Ads Extension version = 2.1.3
2019-01-29 23:11:39 - OpenCart version = 2.3.0.2
2019-01-29 23:11:39 - PHP version = 7.1.20
2019-01-31 0:03:40 - Get Sample product feed
2019-01-31 0:03:40 - Complete - Get Sample product feed
2019-01-31 0:03:43 - Facebook Ads Extension version = 2.1.3
2019-01-31 0:03:43 - OpenCart version = 2.3.0.2
2019-01-31 0:03:43 - PHP version = 7.1.20
`

**Comment by shue249** [comment]
@itshaman Thanks for the information. Can I ask you for some more further details.

1. Can you go to any product in the OpenCart admin product management module. Click on Edit and click on Save. You do not need to make any changes. We want to force the plugin to make a immediate sync for that product to facebook. After which, can you share the log file again, we want to check if there are any errors during this operation.

2. What is the Facebook page that you are using?

3. Can you share the screenshot where you are seeing the message "The Facebook access token is invalid. Please click on Facebook Ads Extension, Manage Settings, go to Advanced options and click on Update token"

**Comment by itshaman** [comment]
Thank you very much for your promptness.
1. Update poduct id=1741. Here is log file
![update_product](https://user-images.githubusercontent.com/16690535/52107612-1840bb00-2608-11e9-8347-3e907521a4a4.PNG)

[2019-02-01_09-50-28_facebook_ads_extension.log](https://github.com/facebookincubator/Facebook-For-OpenCart/files/2820406/2019-02-01_09-50-28_facebook_ads_extension.log)

2. One of facebook page is https://www.facebook.com/VikArtMolb
I am owner of it. Permision is right. Also configured in buisnes.facebook.com

3. Screen with error. I got it when login to admin panel Dashboard of Opencart.
![dashboard](https://user-images.githubusercontent.com/16690535/52107356-0b6f9780-2607-11e9-8976-8c29a259642f.PNG)
When i click to eneter Facebood Ads Extention i got next error
![step_1 go_to_facebook_ads_extetnsion](https://user-images.githubusercontent.com/16690535/52107606-10811680-2608-11e9-8730-56324f7c13b5.PNG)

**Comment by itshaman** [comment]
And one more question.
Where can I edit text about privacy policy in popUp window allowing the use Cookies?

**Comment by shue249** [comment]
@itshaman Thanks for the response. Pls give us some time to look into this in more details.

For the cookie bar text, you will need to manually change the text on your server. There are 2 ways to do this.

1. Change the install.xml file inside the zip file. Then uninstall the extension and reinstall again. This will make the changes be reflected on your server.

The changes are in:
https://github.com/facebookincubator/Facebook-For-OpenCart/blob/master/install.xml#L494-L495
https://github.com/facebookincubator/Facebook-For-OpenCart/blob/master/install.xml#L658-L659

2. Change the modified files in your server. The file should be in <your opencart root folder>/system/storage/modification/catalog/view/<theme>/template/common/header.tpl
![image](https://user-images.githubusercontent.com/11957585/52165463-124bf680-273c-11e9-88e4-4e970fd2e794.png)

**Comment by itshaman** [comment]
Hi
Any issues about "Update token" avalible now?

**Comment by shue249** [comment]
@itshaman We looked into the log file and it seems that the product sync with the access token is running fine. We will be putting in more logging mechanism in our next version and we like to invite you to help us troubleshoot this further with the additional logging, so that we can figure out where is the problem.

We wil update you soonest once the plugin is ready.

**Comment by itshaman** [comment]
I'd be happy to help you make it better
Any logs i can send you

**Comment by shue249** [comment]
@itshaman Thanks for your help, this is valuable.

We did a double check again on your log file and found that there could be something wrong with the settings.

In particular, we found that there are no successful response for the product updates after 12th Jan.
![image](https://user-images.githubusercontent.com/11957585/52608348-9462ba80-2eb4-11e9-9ab7-e2a85653f8bc.png)

Will you be able to help us with some tests below?
1. Will you be able do the below test to verify if the product integration is working properly on your OpenCart site?
a. Create a new product on OpenCart. Check if this product is also created on Facebook catalog.
![image](https://user-images.githubusercontent.com/11957585/52607835-b3f8e380-2eb2-11e9-9dd4-0e6e834e2a5f.png)

b. Make some edits to that new product created, eg changing the description or price. Check if the changes is also reflected on Facebook catalog (you will need to refresh the Facebook catalog page).
![image](https://user-images.githubusercontent.com/11957585/52607909-fe7a6000-2eb2-11e9-97d8-2e3acb1d25f5.png)

c. Delete away that new product. Check if the product has been removed from Facebook catalog (you will need to refresh the Facebook catalog page and search for the product).
![image](https://user-images.githubusercontent.com/11957585/52607961-40a3a180-2eb3-11e9-8b29-f1a43ac7ed99.png)

d. Check the Facebook Ads Extension log file. Verify that the tests we did was recorded as success on the log file.
![image](https://user-images.githubusercontent.com/11957585/52608021-8f513b80-2eb3-11e9-9113-90314142b779.png)

2. Are you familiar with running SQL queries on your database server? If so, can you run this check to verify your access token and catalog id?
a. Run this SQL command on your database server. 
`select` * from oc_setting where `code` = 'facebook';
Check if you can locate the Facebook Ads Extension settings, particularly for page_token, page_id and catalog_id.
![image](https://user-images.githubusercontent.com/11957585/52608505-2ec2fe00-2eb5-11e9-9069-0a713738f986.png)

3. Verifying that the settings are valid using Facebook graph explorer. Access Facebook graph explorer at https://developers.facebook.com/tools/explorer/. 
a. Copy paste in the page_token from your database server to the graph explorer. Copy and paste in the page_id. Click on Submit button and check if you get a success response with information shown.
![image](https://user-images.githubusercontent.com/11957585/52608770-dd673e80-2eb5-11e9-9971-120f930fba0d.png)

b. Repeat the same check but using catalog_id now. 
![image](https://user-images.githubusercontent.com/11957585/52608814-04257500-2eb6-11e9-8a7e-9900f09b2163.png)

Let us know if this makes sense for you

**Comment by shue249** [comment]
@itshaman Wondering if you have performed the tests and if there's any results from your tests?

**Comment by itshaman** [comment]
In Graph Explorer I got error.
Realy I change my pass at FB. 
It happens usualy. FB write me "You page is hacked, lets change password"

How can I renew access token after change pass of my fbpage?

"message": "Error validating access token: The session has been invalidated because the user changed their password or Facebook has changed the session for security reasons.",
    "type": "OAuthException",
    "code": 190,
    "error_subcode": 460,

**Comment by shue249** [comment]
@itshaman In this case, this is a case of the access token being invalidated. Can you share with me a screenshot of the Facebook Ads Extension module on your OpenCart again.

**Comment by shue249** [comment]
@itshaman We have released a latest version of the plugin, v2.1.4. Can you access the Facebook Ads Extension module, download the log file and share the log file with us for us to take a look at the additional logs.

**Comment by itshaman** [comment]
[afterUpdateFBAds.txt](https://github.com/facebookincubator/Facebook-For-OpenCart/files/2901352/afterUpdateFBAds.txt)

Ohh,

> Error with getting the initial product sync status The Facebook access token is invalid. Please click on Facebook Ads Extension, Manage Settings, go to Advanced options and click on Update token.

**Comment by itshaman** [comment]
When I update product
![default](https://user-images.githubusercontent.com/16690535/53360197-f6580100-3945-11e9-9fcc-39f338ed68de.png)
Go to Facebook Ads Extention
![default](https://user-images.githubusercontent.com/16690535/53360278-2b645380-3946-11e9-8055-3eb52abadc78.png)
Enter to Settings
![default](https://user-images.githubusercontent.com/16690535/53360332-5353b700-3946-11e9-9778-70e24c48015d.png)

**Comment by itshaman** [comment]
[lastLog.txt](https://github.com/facebookincubator/Facebook-For-OpenCart/files/2902119/lastLog.txt)

**Comment by itshaman** [comment]
I try to generate token in Graph API Explorer
But There i must to create aplication to give it access to page (((
I have admin permission on both of page/site 
Button "Renew Access Token" will make happy many people, i think.
Facebook usualy ask me to change password - > it broke token to page & stoping work of modules

**Comment by shue249** [comment]
@itshaman Thank you for the feedback. Our plugin has the mechanism to be able to "Renew Access Token", but we are not sure why it did not appear successfully for you.

Alternatively, will you be able to re-do the Facebook Ads Extension setup again? This means for you to delete away the existing settings and complete the setup again? In this case, the access token will be updated as the correct one.

**Comment by itshaman** [comment]
It will delete all my catalog and pixels in buisnes.facebook.com?
After last so peration, at my fb page deleted storage ang all my collection of it.

**Comment by shue249** [comment]
@itshaman Deleting and re-setup pf the Facebook Ads Extension will delete the existing catalog and re-create a new catalog. There is another alternative if you do not wish to redo the setup. The steps are below:
1. Access the Facebook Ads Extension module on your OpenCart admin panel
2. Launch the Javascript console
![image](https://user-images.githubusercontent.com/11957585/53614531-41f3fe80-3c15-11e9-95d1-d5e300ce5ed7.png)
3. Copy, paste and run this code on the javascript console
`window.facebookAdsToolboxConfig.tokenExpired = 'true'`
4. Launch the Facebook Ads Extension popup and you should be able to see the option to update the token.
![image](https://user-images.githubusercontent.com/11957585/53614459-06f1cb00-3c15-11e9-825b-c042eb15cebb.png)

**Comment by bike9527** [comment]
@shue249  I have the same problem  
![image](https://user-images.githubusercontent.com/32729137/54341870-f52d1080-4675-11e9-821b-b37a48ad8dd5.png)

**Comment by shue249** [comment]
@bike9527 Can you click on the "Manage settings" button and you will be able to reset the access token via the Advanced options.

**Comment by chicho-picho** [comment]
Hi there, i have absolutely the same issue from first post. When I go to Facebook Ads Extentions i have "error" and when i go to Manage Settings, than Advanced Settings, can't find where to Update Token? I'm using ver. 2.1.5 on OC 3.0.2

**Comment by shue249** [comment]
Everyone, we have just released a version 2.1.6 which will solve the "error" or "Bad request" error message, allowing you to update the token and without the need to run any codes on the console. Please update to the latest version here - https://github.com/facebookincubator/Facebook-For-OpenCart/releases/tag/v2.1.6

However, if you still encounter the "Invalid access token" even after updating the token, pls comment in this ticket https://github.com/facebookincubator/Facebook-For-OpenCart/issues/144 and we will follow up there.

Closing off this ticket for now.

**Comment by VitaliiFesh** [comment]
Hello, I've just updated to 2.1.6 and still getting same error. 
Please advice. 
Logs:
2019-04-17 19:44:12 - Facebook Ads Extension version = 2.1.6
2019-04-17 19:44:12 - OpenCart version = 2.3.0.2
2019-04-17 19:44:12 - PHP version = 5.6.40
2019-04-17 19:44:12 - Verifying Availability of Facebook Ads Extension Settings Availability
2019-04-17 19:44:12 - FAE Setting ID = true
2019-04-17 19:44:12 - Pixel ID = true
2019-04-17 19:44:12 - Pixel use PII = true
2019-04-17 19:44:12 - Catalog ID = true
2019-04-17 19:44:12 - Page ID = true
2019-04-17 19:44:12 - Page Token = true
2019-04-17 19:44:12 - Feed ID = true
2019-04-17 19:44:12 - Upload ID = true
2019-04-17 19:44:12 - Upload end time = true
2019-04-17 19:44:12 - Messenger plugin = false
2019-04-17 19:44:12 - Messenger plugin, greeting text code customization = false
2019-04-17 19:44:12 - Messenger plugin, locale customization = false
2019-04-17 19:44:12 - Messenger plugin, theme color code customization = false

**Comment by alexob** [comment]
Hi,
I have the same problem as in 1st post. (version 2.1.6)

**Comment by shue249** [comment]
@VitaliiFesh , @alexob 

Launch the Facebook Ads Extension popup and you should be able to see the option to update the token.
![image](https://user-images.githubusercontent.com/11957585/53614459-06f1cb00-3c15-11e9-825b-c042eb15cebb.png)

Click on Update token. If the error message still remains, can you leave a comment on this ticket. We have a small group of users who still have problems with the updated access token which we are checking through.

https://github.com/facebookincubator/Facebook-For-OpenCart/issues/144

**Comment by robihdr** [comment]
@shue249 Thanks

**Comment by SamairZia** [comment]
Hey,
I have updated the version of this plugin to v2.1.6 , but still it shows the error "The Facebook access token is invalid. Please click on Facebook Ads Extension, Manage Settings, go to Advanced options and click on Update token.". But no such button was found there I'm using Opencart version 2.3.0.2.
I have ran this code on the JavaScript console "window.facebookAdsToolboxConfig.tokenExpired = 'true'" after running this code it shows the update token button and Key is updated and everything works fine. But once i logged out of my Facebook page (that is connected with my opencart site) the error comes back. 
@shue249

**Comment by chicho-picho** [comment]
Hi there, i had the same problem with 2.1.5, i Updated token and after that i have no problems. I think the problem is very simple.... you have 2 or more facebook profiles and when you switch to other profile you get the error. It's very simple :)

**Comment by shue249** [comment]
Everyone, thank you again for your patience and your information to help us look into this case. We will like to take this chance to share what have we fixed, how should you go about resolving the problem on your OpenCart and what should you do if the access token still shows up as invalid. Please refer to this comment for more details. https://github.com/facebookincubator/Facebook-For-OpenCart/issues/144#issuecomment-489399216

If you are still encountering error, pls kindly help us to post your screenshot of the error message (pls ensure that you are on version 2.1.7 plugin) on this separate ticket #158. This will greatly help us to clearly separate out the new issues so that we can focus on the problems.

**Comment by tksir** [comment]
![Untitled](https://user-images.githubusercontent.com/54794143/65016782-b50b5b00-d942-11e9-897f-ebcfde280b31.png)
I went to Advanced Setting and tried updating. It takes a long time but doesn't update.

**Comment by tksir** [comment]
![image](https://user-images.githubusercontent.com/54794143/65018364-8becc980-d946-11e9-9cec-c7ba70fda67d.png)
And Advanced Options don't have an Update Token.

**Comment by tksir** [comment]
![Untitled](https://user-images.githubusercontent.com/54794143/65018749-73c97a00-d947-11e9-9f5f-3551a8525b85.png)
And here link is not shown well.

**Comment by tksir** [comment]
![Untitled](https://user-images.githubusercontent.com/54794143/65111057-25c07f00-d9f8-11e9-859d-c971014fc79c.png)
When I try to fetch the products it's being always processed but never completes and finishes.

**Comment by Stiliyana1** [comment]
Hi, I have the same problem with the products. Its being always processes and not updated in my fb page 
![ggfgfgfgf](https://user-images.githubusercontent.com/64435880/80587464-471e2a80-8a1f-11ea-87a9-a41ad51e4654.JPG)
![gfgdg](https://user-images.githubusercontent.com/64435880/80587468-47b6c100-8a1f-11ea-936d-6e4c7c249a45.JPG)
